core/blockchain.go

```diff
@@ -1228,6 +1232,28 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals bool) (int, []
 			bc.reportBlock(block, nil, ErrBlacklistedHash)
 			return it.index, events, coalescedLogs, ErrBlacklistedHash
 		}
+		// If the block is known (in the middle of the chain), it's a special case for
+		// Clique blocks where they can share state among each other, so importing an
+		// older block might complete the state of the subsequent one. In this case,
+		// just skip the block (we already validated it once fully (and crashed), since
+		// its header and body was already in the database).
+		if err == ErrKnownBlock {
+			log.Debug("Inserted known block", "number", block.Number(), "hash", block.Hash(),
```

Comment:
Since this should only ever happen on Clique, perhaps we might want to check that it's indeed Clique, and file a `Warn` about this otherwise? 
Just to make sure that we notice if this assumption ever breaks?

Reply:
Done.